### PR TITLE
Fixes #904 User can choose resolution for the image

### LINKED ISSUE #904: Let the user choose the resolution of images he/she sends
Newer cameras can produce images up to 20MB large.
Let the user choose the resolution of images he/she sends on mobile networks, with a minimum, to ensure the image is legible.

### PR DESCRIPTION
**Pull request by vaishali51**
## Description

User can choose resolution for the image if mobile data is ON so that data can be saved.

## Related issues and discussion
#904 
 ## Screen-shots, if any
screenshot_2018-03-05-16-25-02-038_openfoodfacts github scrachx openbeauty

### CODE REVIEW

**Comment by Karljoones** [comment]
@teolemon should this be a setting in the preferences? Allow them to set the resolution that they want to upload in the preferences. This would probably get annoying to request this for each image, 3 times per product upload.

**Comment by teolemon** [comment]
Yes, that should definitely be in settings. Also a nice addition would be to give an approximation of the image size produced, and the approximate time it takes on 2G, 3G, 4G

**Comment by vaishali51** [comment]
Ok, I'll update the code then.

**Comment by vaishali51** [comment]
Update

User can choose resolution in settings.

screenshot_2018-03-06-15-07-10-641_openfoodfacts github scrachx openbeauty

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java [hunk 1]
```diff
@@ -396,50 +406,122 @@ private void onPhotoReturned(File photoFile) {
...
+                                public void onSuccess() {
+                                    mAttacherimgSaveFront.update();
+                                }
+
+                                @Override
+                                public void onError() {
+                                }
+                            });
+                } catch (Exception e){
+                    Toast.makeText(getApplicationContext(), "Please set Resolution First", Toast.LENGTH_SHORT).show();
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java]: This needs to be translateable, and if it is not set, it should have a default setting

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java [hunk 2]
```diff
@@ -396,50 +406,122 @@ private void onPhotoReturned(File photoFile) {
...
+                                public void onSuccess() {
+                                    mAttacherimgSaveNutrition.update();
+                                }
+
+                                @Override
+                                public void onError() {
+                                }
+                            });
+                } catch (Exception e){
+                    Toast.makeText(getApplicationContext(), "Please set Resolution first", Toast.LENGTH_SHORT).show();
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java]: Same as above

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java [hunk 3]
```diff
@@ -396,50 +406,122 @@ private void onPhotoReturned(File photoFile) {
...
+                                public void onSuccess() {
+                                    mAttacherimageSaveIngredients.update();
+                                }
+
+                                @Override
+                                public void onError() {
+                                }
+                            });
+                } catch (Exception e){
+                    Toast.makeText(getApplicationContext(), "Please set Resolution first", Toast.LENGTH_SHORT).show();
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java]: Same as above

File: app/src/main/res/xml/preferences.xml [hunk 4]
```diff
@@ -33,6 +33,14 @@
             android:summaryOn="@string/preference_enable_mobile_data_summary_on"
             android:title="@string/preference_enable_mobile_data_title" />
 
+        <ListPreference
+            android:defaultValue="640 X 480"
+            android:dialogTitle="Choose Resolution"
+            android:key="ImageUpload"
+            android:selectable="true"
+            android:summary="@string/prefs_image_upload"
+            android:title="Choose resolution" />
```
Comment (Karljoones) [inline app/src/main/res/xml/preferences.xml]: Needs to be translatable, and something along the lines of "Image upload resolution"

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java (hunk 2 above)
Comment (teolemon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java]: Please set a resolution first

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java (hunk 3 above)
Comment (teolemon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java]: "Please set a resolution first"

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java (hunk 1 above)
Comment (teolemon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/SaveProductOfflineActivity.java]: "Please set a resolution first"

**Comment by teolemon** [comment]
@vaishali51 tiny conflict to fix.

**Comment by teolemon** [comment]
@vaishali51 Can you add a screenshot of the current version ?
@Karljoones

**Comment by vaishali51** [comment]
I have added the keys in Translation Editor, but I am facing trouble in the output.
img_20180310_202344
screenshot_2018-03-10-20-25-33-996_openfoodfacts github scrachx openbeauty

**Comment by teolemon** [comment]
Nice 👍  The build is failing:
```org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:compileObfDebugJavaWithJavac'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed; see the compiler error output for details.
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:51)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:36)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:99)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:52)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:37)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:35)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:25)
	at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:198)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:183)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:120)
	at com.android.build.gradle.tasks.factory.AndroidJavaCompile.compile(AndroidJavaCompile.java:95)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$IncrementalTaskAction.doExecute(DefaultTaskClassInfoStore.java:173)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
	... 27 more```

**Comment by teolemon** [comment]
Merging. Thanks @vaishali51 :-)

**Pull request merged**